### DISCUSSION TITLE
3.4.2 括弧等が連続した場合の処理

### DISCUSSION THREAD

**Discussion by kidayasuo** [w3c/jlreq-d#61] [jlreq-d-drafts]
[3.4.2 括弧等が連続した場合の処理](https://github.com/w3c/jlreq-d/blob/5a5fbfeed7d426234d2ca22ce5ff9c21abb589ff/drafts/3_行の組版処理.md#342括弧等が連続した場合の処理)の議論

**Comment by kidayasuo** [comment]
@KobayashiToshi 

ここの説明ですが、ソフトウェア側にとってフォントは与えられたもので、括弧グリフの幅や、その中のアキが1/2かどうかなどは関与範囲外、という事情があります。JLReqのようにこれらが半角という前提を持つ場合はフォントのことを考える必要はなかったのですが。

ので、次の二段階のロジックで説明するのはどうでしょう？

#### 1. 括弧や句読点のグリフがどうあるべきか
- 和文括弧や句読点は通常、全幅の幅を持つ
- 和文括弧や句読点のグリフには、（区切りを示すために）通常その字幅の半分の大きさのアキが組み込まれている。開き括弧は前側、閉じ括弧や句読点は後ろ側、中点は前後に1/4づつ
- 上二つに通常とつけたように、そこから外れたデザインもあるかもしれない。外れる場合にはタイプフェースデザイナーが責任を持ってバランスを考えて設計する（MS P なんたらはどうでしょうね）。
- そのアキを削除するための機能がフォントにある（halt, chws。なぜかというと、字幅の½とは限らないので）

#### 2. そのグリフをどう扱うか
- 通常は、それぞれの字幅に従って並べる（ここまでが3.4.1）
- 約物が連続した場合には、不要なアキができるので、削除する必要がある。その組み合わせや方法を述べる。この段階では、1で、アキの典型的な大きさや、その削り方を説明しているので、全幅やその半分などの概念に触れず、どのような文字の組み合わせで削除をすれば良いかを説明する（3.4.2）

**Comment by KobayashiToshi** [comment]
だいたい木田さんの指摘された方向で修正いたしました．ご検討ください．

それから，“約物”や“アキ”という用語は使用しないようにしています．特に“アキ”は，JRReqで定義された用語ですので，ここに使用するのは望ましくない．ここでの“句読点の後ろにあるアキ”という説明は，全幅の字幅の後ろにあるアキという意味になる恐れがある．

基本的な考え方として，“字幅”“字詰め方法”“行送り方向”など，まあ，日常の感覚でなんとか理解できることばは使用するが，“本文”や“アキ”など定義を読んで，その意味を理解したうえでないと誤解を与える用語は使用しない．その意味で“約物”も意味があいまいな用語です（記述記号に限定する考え方と，その他一般の記号も約物に含める考え方とがある）ので，使用しないようにした．

ところで，用語の定義はどうしますか？　どうしても必要となる用語だけ，初出で定義する，ということでいいのかな．JRLeqみたいに用語集できちんと定義するのもありですが，ちょっと面倒．

**Comment by himorin** [comment]
closing per https://github.com/w3c/jlreq-d/pull/132
check current draft at https://github.com/w3c/jlreq-d/blob/gh-pages/drafts/%E5%AE%9F%E8%A3%85%E7%B7%A8%EF%BC%9A%E8%A1%8C%E3%81%B8%E3%81%AE%E6%96%87%E5%AD%97%E3%81%AE%E9%85%8D%E7%BD%AE.md#333%E6%8B%AC%E5%BC%A7%E7%AD%89%E3%81%8C%E9%80%A3%E7%B6%9A%E3%81%97%E3%81%9F%E5%A0%B4%E5%90%88%E3%81%AE%E5%87%A6%E7%90%86
